test/sqlsmith/mzcompose.py

```diff
@@ -48,14 +49,18 @@ def __init__(
         if restart:
             config["restart"] = restart
 
+        if memory:
+            config["deploy"] = {"resources": {"limits": {"memory": memory}}}
+
         super().__init__(name=name, config=config)
 
 
 MZ_SERVERS = [f"mz_{i + 1}" for i in range(4)]
 
 SERVICES = [
     # Auto-restart so we can keep testing even after we ran into a panic
-    StandaloneMaterialized(name=mz_server, restart="on-failure")
+    # Limit memory to prevent long hangs on out of memory
+    StandaloneMaterialized(name=mz_server, restart="on-failure", memory="48GB")
```

Comment:
I think this needs to be 48 / 4 (actually you may wish to make the `4` a constant or a command-line argument now that it will be used in multiple places.

Alternatively, just hard-code 8Gb for each and call it a day.